### ISSUE TITLE
Unable to see or modify some repository settings, e.g. "extra borg arguments"

### ISSUE THREAD

**Issue by goebbe** [borgbase/vorta#1963]
#### Description
Some important **repository setting** can only be seen and edited in the UI when a new repo is initialised.

Example: 
Depending on the backup server, "extra borg arguments" are required for a repository.
Examples for the use of "extra borg arguments": 
  - on synology (home) servers, the `--remote-path` argument is required.
  - if different versions of borg are provided on the backup server, the --remot-path has to be adapted.

When **adding a new repository**, the advanced tap offers a field for "extra borg arguments" in the UI - this works well. 

Currently, I see two related **issues** with the "extra borg arguments" in Vorta: 
a) Once the repo is initialized, the "extra borg arguments" for the repository **cannot be seen** in the UI. 
b) After initializing, it is currently **not possible to edit/ change** the "extra borg arguments" for an existing repo in the UI. 

#### Expected behaviour / possible enhancements:
All the repo settings and options should be visible. 
The user should be able to modify the setting.

In our example: 
for a) The "extra borg arguments" for a repo  could be shown in the UI, e.g. on the right side (Info field) of the  repository tab
for b) Allow to change/ edit the "extra borg arguments" for a repository

#### Workarounds: 
for a) - i.e. "extra borg arguments" cannot be seen in the UI
    Select the profile
    Export settings of the profile / repo (using Vorta) to a .json file
    Open the .json file to see the "extra borg arguments" for the repository
    Delete the .json file

for b) - i.e. cannot change "extra borg arguments" in Vorta
Create a new repository, with different "extra borg arguments"
or: 
Export settings to a .json-file, modify the setting using a text editor, rename and re-import the settings using the .vort-init.json file. 

#### Environment
- OS: Linux
- Vorta version: 0.9.1. and 0.10.3
- Installed from: pip3
- Borg version: 1.2.7 and 1.4

**Comment by shivansh02** [comment]
Do any of these look good?
![extraborg-1](https://github.com/borgbase/vorta/assets/89853707/ce1e7e1f-dedc-4ce8-80de-fd3d8d516739)
![extraborg-2](https://github.com/borgbase/vorta/assets/89853707/4c6d2216-bc04-4687-a24a-355c564a3a32)

Original:
![Screenshot from 2024-03-13 00-18-34](https://github.com/borgbase/vorta/assets/89853707/232e4c02-d069-47e6-b9dd-ab9b7bdb8353)

**Comment by goebbe** [comment]
@shivansh02  Thank you for the mock-ups! 
For **showing** the "extra borg argument": 
That is exactly how I imagined it. A new line with information under the Encryption information. 

With respect to the **user interface**, I would reserve the right field for information only. 
Instead one could put a new "edit icon" (from your first picture), next to the "Copy to clipboard" icon. 
I guess that @m3nu should take a look at this.  

Note that another repo-setting that could be editable, via this edit button could be the "Repository Name" - which, at the moment, cannot be changed after the setup.

**Comment by shivansh02** [comment]
> Instead one could put a new "edit icon" (from your first picture), next to the "Copy to clipboard" icon.

You're right, that is a much better place to put it. I'll send an updated mockup with this and the Edit Repo Dialog once @m3nu gives his inputs.

**Comment by shivansh02** [comment]
Related to #1663.

**Comment by goebbe** [comment]
It's unfortunate that issues are automatically closed, on the basis that nobody is currently working on the issue. 
There is a description and a proposal for the UI - what is missing is a "thumbs up" from the core developers or a decision. 

Some projects also use a "good first issue" tag, to signal that 
a) this is a real issue that could be fixed and 
b) anybody who is willing and able to fix is encouraged to work on the issue.

**Comment by m3nu** [comment]
We can leave it open, if there is interest.

**Comment by goebbe** [comment]
With Vorta 0.10.1 once a new repo with extra borg arguments has been created, there is no way to _see_ or _edit_ these extra arguments - this can be inconvenient when you have to check/ change/ adapt the borg arguments.

Currently, the workaround would be to create a new repo (with different extra arguments). :-(

Perhaps Vorta could implement something similar to Schedule > Shell commands, where one can always see/ edit the extra borg arguments?  

If this is considered to be an issue that should be fixed,  this should stay open.

**Comment by goebbe** [comment]
While this issue is about "extra borg arguments" at the level of the repository, the same applies to other repository settings - that are set in the "Add New Repository"-dialogue. 

Currently, once the repo has been added, there are no easy means to see all the settings - and the user cannot easily check or change the repo-settings. 
One _workaround_ is to again "Add New Repository" for the same repo, with changed settings. 

In Vorta (0.10.3) only the following repo-setting can be seen and changed:  
- compression 
- SSH Key setting 

Here is a **list of the repo settings** that cannot be changed via the Vorta GUI, after the initial setup: 
- Repository **URL** (can be seen, but not be changed) 
- **Repository Name** (can be seen, but not be changed) 
- ~~Passphrase (cannot be changed)~~ 
- **Encryption** (cannot be seen and not be changed)
- **Extra Borg Arguments** (cannot be seen and not be changed)

Edit: changing the passphrase of a repository is now supported (Vorta 0.11)

**Comment by ispmarin** [comment]
Editing configurations would be helpful, especially --remote-path configurations for borg. Even if it's not available on the GUI, is there a way to change it via CLI, in a configuration file?

**Comment by goebbe** [comment]
The following workaround can be used: 

**1. Export the current settings** (per profile): 
  - In the left pane, select a profile and choose "Export current profiles" at the left bottom.
  - As a result, a `profilename.json` file will be written to your home directory. 

**2. Edit the resulting .json-file**
  - Use a Text editor of your choice to view and modify the settings.

**3. Rename and re-import the .json-file** with the new settings
To re-import the .json file with the setting: 
  - The json-file has to be placed in the users home directory ~
  - it has to be re-named `.vorta-init.json`
  - this .json-file will be imported automatically when starting (or restarting) Vorta
  - this .json file will be deleted after import.
  - If the name of the profile already exists: the old profile will be replaced by the new settings

**Comment by goebbe** [comment]
> The following workaround can be used:
> 
> **1. Export the current settings** (per profile):
> 
>     * In the left pane, select a profile and choose "Export current profiles" at the left bottom.
> 
>     * As a result, a `profilename.json` file will be written to your home directory. This file contains the settings of a profile in a human-readable form.
> 
> 
> **2. Edit the resulting .json-file**
> 
>     * Use a Text-editor to view and modify the settings in the .json file. E.g. You can check or change the settings of the `--remote-path` configuration.
> 
> 
> **3. Rename and re-import the .json-file** with the new settings To re-import the .json file with the setting:
> 
>     * The json-file has to be placed in the users home-directory ~
> 
>     * it has to be re-named to `.vorta-init.json`
> 
>     * this .json-file will be imported automatically when starting (or restarting) Vorta
> 
>     * this .json-file will be deleted after import.
> 
>     * If the _name of the profile_ already exists, the settings of the existing profile will be replaced by the new settings

**Comment by goebbe** [comment]
Another workaround, on Linux, is the use of the "**DB Browser for sqlite**" aka "**sqlitebrowser**" to edit the database (DB) with the Vorta settings directly:

1. Install the DB Browser / sqlitebrowser
2. Quit Vorta completely
3. Open `~/.local/share/Vorta/settings.db`, using the DB Browser. 
     If you use the _Flatpak_ build, open: `~/.var/app/com.borgbase.Vorta/data/Vorta/settings.db`
4. Choose the "Browse data" tab, select the table "repomodel", edit "exta_borg_arguments", Apply
5. Close DB Browser and restart Vorta

Disclaimer: Use with care, since this will edit the SQLite settings.db of Vorta directly. So there is no sanity checking of any kind.
Just in case: Make a copy of settings.db before editing. ;-)

**Comment by pedrompcaetano** [comment]
Simply adding the extra arguments didn't solve it for me.
After struggling with this for way too long, I decided to delete the flatpak files and reimport the .json backup with the --remote--path set in extra_borg_commands. (rm ~/.var/app/com.borgbase.Vorta)
Issue solved! Rsync.net remote backups and related borg operations are now working properly :))

Hope this helps other users.
Thank you for your time working on this great project!
